sdk/inc/azure/core/az_context.h

```diff
@@ -61,25 +60,24 @@ extern az_context az_context_application;
 /**
  * @brief Creates a new expiring #az_context node that is a child of the specified parent.
  *
- * @param[in] parent The #az_context node that the new node is to be a child of; passing `NULL` sets
- * the parent to #az_context_application.
+ * @param[in] parent The #az_context node that the new node is to be a child of.
  * @param[in] expiration The time when this new child node should be canceled.
  *
  * @return The new child #az_context node.
  */
 AZ_NODISCARD AZ_INLINE az_context
 az_context_create_with_expiration(az_context const* parent, int64_t expiration)
 {
-  return (az_context){ ._internal
-                       = { .parent = ((parent != NULL) ? parent : &az_context_application),
-                           .expiration = expiration } };
+  _az_PRECONDITION_NOT_NULL(parent);
```

Comment:
Add another precondition that `expiration` is `>= 0`

Reply:
On second though, should we allow a creation of an already expired context, or should the precondition be `expiration > 0`?

Reply:
Setting it to 0 is the equivallent of cancelling it.  I see no reason to not allow 0, its really no different than 1.  They'll both be expired before they could ever be used.  I'd say the same for values less than zero, though there is an argument that could be made that these values should be positive.